packages/core/src/history.ts

```diff
@@ -76,29 +76,51 @@ export const writeHistory = async (historyPath: string, data: HistoryDataPoint)
 };
 
 export class AllureLocalHistory implements AllureHistory {
-  constructor(private readonly historyPath: string) {}
+  constructor(
+    private readonly params: {
+      historyPath: string;
+      limit?: number;
+    },
+  ) {}
 
   async readHistory() {
-    const path = resolve(this.historyPath);
+    const path = resolve(this.params.historyPath);
 
     try {
-      return (await readFile(path, { encoding: "utf-8" }))
+      const historyPoints = (await readFile(path, { encoding: "utf-8" }))
         .split("\n")
         .filter((line) => line && line.trim() !== "")
         .map((line) => JSON.parse(line) as HistoryDataPoint);
+
+      if (this.params.limit) {
+        return historyPoints.slice(-this.params.limit);
+      }
+
+      return historyPoints;
     } catch (e) {
       if (isFileNotFoundError(e)) {
         return [];
       }
+
       throw e;
     }
   }
 
   async appendHistory(data: HistoryDataPoint) {
-    const path = resolve(this.historyPath);
+    const path = resolve(this.params.historyPath);
     const parentDir = dirname(path);
 
     await mkdir(parentDir, { recursive: true });
-    await writeFile(path, `${JSON.stringify(data)}\n`, { encoding: "utf-8", flag: "a+" });
+
+    if (!this.params.limit) {
+      await writeFile(path, `${JSON.stringify(data)}\n`, { encoding: "utf-8", flag: "a+" });
+      return;
+    }
+
+    const existingHistory = await this.readHistory();
+    const updatedHistory = [...existingHistory, data].slice(-this.params.limit);
+    const fileContent = updatedHistory.reduce((acc, point) => `${acc}${JSON.stringify(point)}\n`, "");
```

Comment:
The same concerns about hitting the max string length apply here. Instead of concatenating a single giant string, we should do the following:

  - read line-by-line and store the lines in an array separately
  - apply the limit
  - write the lines to the file
  - append the new entry